test/replication-luatest/gh_7253_election_long_wal_write_test.lua

```diff
@@ -59,12 +59,39 @@ local function server_wait_wal_is_blocked(server)
                        check_wal_is_blocked_f)
 end
 
+local function server_set_replication(server, replication)
+    server:exec(function(replication)
+        box.cfg{replication = replication}
+    end, {replication})
+end
+
+-- Allow WAL writes one by one until the synchro queue becomes busy. The
+-- function is needed, because during box.ctl.promote() it is not known for sure
+-- which WAL write is PROMOTE - first, second, third? Even if known, it might
+-- change in the future.
+local function play_wal_until_synchro_queue_is_busy(server)
+    t.helpers.retrying({}, server.exec, server, function()
+        if not box.error.injection.get('ERRINJ_WAL_DELAY') then
+            error('WAL did not reach the delay yet')
+        end
+        if box.info.synchro.queue.busy then
+            return
+        end
+        -- Allow 1 more WAL write.
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+        error('Not busy yet')
+    end)
+end
```

Comment:
Let's move this to luatest_helpers. It's quite obvious that this function is needed pretty often:
```
s.petrenko@spetrenko:~/Source/sptnt$ grep -r 'function play_wal_until_synchro_queue_is_busy' ./test
./test/replication-luatest/gh_7253_election_long_wal_write_test.lua:local function play_wal_until_synchro_queue_is_busy(server)
./test/replication-luatest/gh_7086_box_issue_promote_assert_test.lua:local function play_wal_until_synchro_queue_is_busy(server)
./test/replication-luatest/gh_6033_box_promote_demote_test.lua:local function play_wal_until_synchro_queue_is_busy(server)
./test/replication-luatest/gh_6842_qsync_applier_order_test.lua:local function play_wal_until_synchro_queue_is_busy(server)
```

Reply:
Thanks, done in a separate commit.